packages/chat-component-bindings/src/messageThreadSelector.ts

```diff
@@ -173,15 +178,14 @@ export const messageThreadSelector: MessageThreadSelector = createSelector(
            * we compare if the readon timestamp is later than the message A sent time
            * if last read message id is not equal to message A's id, check the read on time stamp.
            * if the last read message is read after the message A is sent, then user should have read message A as well */
-
-          let readNumber = 0;
+          const messageReadByNames: { id: string; name: string }[] = [];
           for (const k in readReceiptForEachSender) {
             const messageid = readReceiptForEachSender[k]['lastReadMessage'];
             const readTime = readReceiptForEachSender[k]['readOn'];
             if (messageid === message.id) {
-              readNumber += 1;
+              messageReadByNames.push({ id: k, name: readReceiptForEachSender[k]['name'] });
```

Comment:
the `if` and `if else` execution blocks are identical, combine into one `if`:
`if (messageid === message.id || (new Date(readTime) >= new Date(message.createdOn))) {

Better, this whole for block could be written in a more functional programming manner for better clarity:
```ts
// Get the names and IDs of everyone who has read this message
const messageReadByList: { id: string; name: string }[] =
  // For each read receipt
  Object.entries(readReceiptForEachSender)
  // Filter to only read receipts that match the message and were read after the message had been created
  .filter(([_, readReceipt]) => readReceipt['lastReadMessage'] === message.id && new Date(readReceipt['readOn']) >= new Date(message.createdOn))
  // Map properties to useful array
  .map(([id, readReceipt]) => ({ id, name: readReceipt['name'] }));
```
(^ note: just a sample possibilty, this was typed in the github comment so copy-pasting it may need syntax fixes to it)

Reply:
Then to reduce code all being in one function/file, this could be moved to a helper function in a seperate file:
```ts
const getParticipantsWhoHaveReadMessage(message, allReadReceipts) => 
  Object.entries(readReceiptForEachSender)
  .filter( etc...
;
```